primer/brand
# Fix checkbox input validation positioning

## Summary

Ref: https://github.com/primer/brand/issues/203

This PR fixes a bug with the positioning of the validation message for checkbox inputs inside a FormControl component.

## List of notable changes:

Styling and layout updates:

* [`packages/react/src/forms/FormControl/FormControl.module.css`](diffhunk://#diff-66f88db62ba93924cf6de6ae216861c98149f0122e02c93173b5778c43cd4b57L13-R14): Changed `.FormControl--checkbox` to use a grid layout instead of flex and added `.FormControl-validation-checkbox` to span columns. [[1]](diffhunk://#diff-66f88db62ba93924cf6de6ae216861c98149f0122e02c93173b5778c43cd4b57L13-R14) [[2]](diffhunk://#diff-66f88db62ba93924cf6de6ae216861c98149f0122e02c93173b5778c43cd4b57R117-R120)
* [`packages/react/src/forms/Checkbox/Checkbox.module.css`](diffhunk://#diff-73b35ce349fc37e97410e2b629ab71dc02d3268812d5e4eed53fa33aa18c3ad7R6): Added `grid-row: 1` to `.Checkbox-wrapper`.
* [`packages/react/src/forms/Radio/Radio.module.css`](diffhunk://#diff-cebde5d1553a837694f74ac3f39e23d3cecbdac76855ed93e741be0df483ed5dR11): Added `grid-row: 1` to the radio button styling.

Storybook updates:

* [`packages/react/src/forms/FormControl/FormControl.stories.tsx`](diffhunk://#diff-27807d60e3cf04e8fc7c732bc664f14b6ee97086aad369c1d6accd00ea7c2bc8R337-R343): Added a new story `ErrorValidationWithCheckbox` for error validation with checkboxes.

## What should reviewers focus on?

Are there any unintended regressions this might cause? 

I looked at the current storybook example form here, as well as the checkbox and radio storybook examples and they all seem to have the intended behavior after my changes, but let me know if I missed anything! 

## Steps to test:



1. Open the FormControl component in CI-deployed preview environment
2. Go to  'w/ Checkbox with Error Validation - Playground' in Storybook
3. Verify that the validation now appears below the checkbox instead of next to them. 

## Supporting resources (related issues, external links, etc):

- Related issue: https://github.com/primer/brand/issues/203
- Previous PR that attempted to fix this issue: https://github.com/primer/brand/pull/513

## Contributor checklist:


## Reviewer checklist:


## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">
<img src="https://github.com/user-attachments/assets/1f3cd2b3-25eb-4ce7-a3ea-d261f7360241"/>
 </td>
<td valign="top">
<img src="https://github.com/user-attachments/assets/101076d6-4cc7-4d22-89bb-fc883df22047"/>
</td>
</tr>
</table>
